councilmatic/settings.py

```diff
@@ -1,38 +1,37 @@
-"""
-Django settings for councilmatic project.
-
-Generated by 'django-admin startproject' using Django 1.8.3.
-
-For more information on this file, see
-https://docs.djangoproject.com/en/1.8/topics/settings/
-
-For the full list of settings and their values, see
-https://docs.djangoproject.com/en/1.8/ref/settings/
-"""
-
-# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 import os
-from .settings_deployment import *
-from .settings_jurisdiction import *
-
+import types
+from django.core.exceptions import ImproperlyConfigured
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
+if os.environ.get('APPLICATION_CONFIG'):
+    filename = os.environ['APPLICATION_CONFIG']
+else:
+    filename = os.path.join(BASE_DIR, 'councilmatic',  'settings_deployment.py')
+
+config = types.ModuleType('config')
+config.__file__  = filename
 
-# Quick-start development settings - unsuitable for production
-# See https://docs.djangoproject.com/en/1.8/howto/deployment/checklist/
+with open(filename, 'rb') as config_file:
+    exec(compile(config_file.read(), filename, 'exec'), config.__dict__)
 
 
+def getSettingValue(setting, required=True):
```

Comment:
Why is the approach necessary?

Reply:
I wanted to be able to raise an exception of one of the settings was not set. I could have just parsed the local settings file up above and then let it fail when an expected setting was not set in there but I thought this would be a bit more friendly. Also, for some reason Django does not allow you to add to the configuration of your app with an environmental variable. You can replace the settings file in it's entirety this way but there's no modular option that will just append (or override) settings from an environmental variable. So, this is what I came up with to keep the secrets secret using the Docker `secrets` thing and to keep things relatively modular.

Reply:
OK, I'll take a look at putting all the settings we need into one file. I was mainly just attempting to get things working more or less the way they are working right now without modifying too much. But I suppose putting everything together is as much a modification as what I've done so far.